pkg/schema/schema.go

```diff
@@ -143,10 +158,63 @@ func newCollectionItemValueType(collectionItemValue interface{}) (yamlmeta.Type,
 	return nil, fmt.Errorf("Collection item type did not match any known types")
 }
 
+func DefaultDataValues(doc *yamlmeta.Document) (*yamlmeta.Document, error) {
+	docCopy := doc.DeepCopyAsNode()
+	for _, value := range docCopy.GetValues() {
+		setDefaultValues(value)
+	}
+
+	return docCopy.(*yamlmeta.Document), nil
+}
+
+func setDefaultValues(node interface{}) {
```

Comment:
Just some food for thought
Having generic `interface{}` in a couple of these methods feels off. Personally, I would like to have the compiler help to ensure we are using the correct types here.
Maybe having an interface called `yamlmeta.Element` or `yamlmeta.Node` or `yamlmeta.Item` not sure naming is hard.

Reply:
Thanks, we are now using the `yamlmeta.Node` type instead of an empty interface.